src/core/ad-sizes.ts

```diff
@@ -371,6 +382,38 @@ const slotSizeMappings: SlotSizeMappings = {
 
 const getAdSize = (size: SizeKeys): AdSize => adSizes[size];
 
+/**
+ * Finds the ad sizes that will be used for a breakpoint given a size mapping
+ *
+ * If ad sizes are defined in the size mapping for the specified breakpoint, we use that.
+ * If no sizes are defined for the breakpoint, use the next smallest breakpoint with defined ad sizes.
+ * If no smaller breakpoints have defined ad sizes, return an empty array
+ */
+const findAppliedSizesForBreakpoint = (
+	sizeMappings: SizeMapping,
+	breakpoint: Breakpoint,
+): AdSize[] => {
+	if (!isBreakpoint(breakpoint)) {
+		return [];
+	}
+
+	let breakpointIndex: 0 | 1 | 2 | 3 = breakpoints.findIndex(
```

Comment:
Really not happy with this type. If we add another breakpoint to the `breakpoints` array we will need to update this type here. I couldn't find another way to get around the `noUncheckedIndexedAccess` typescript rule, so please shout if you have an idea!

Run `yarn tsc --project ./tsconfig.core.json` locally to test.